record/program/src/state.rs

```diff
@@ -0,0 +1,90 @@
+//! Program state
+use {
+    borsh::{BorshDeserialize, BorshSchema, BorshSerialize},
+    solana_program::{program_pack::IsInitialized, pubkey::Pubkey},
+};
+
+/// Struct wrapping data and providing metadata
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema, PartialEq)]
+pub struct AccountData {
+    /// Struct version, allows for upgrades to the program
+    pub version: u8,
+
+    /// The account allowed to update the data
+    pub authority: Pubkey,
+
+    /// The data contained by the account, could be anything serializable
+    pub data: Data,
+}
+
+/// Struct just for data
+#[derive(Clone, Debug, Default, BorshSerialize, BorshDeserialize, BorshSchema, PartialEq)]
+pub struct Data {
+    /// The data contained by the account, could be anything or serializable
+    pub bytes: [u8; Self::DATA_SIZE],
+}
+
+impl Data {
+    /// very small data for easy testing
+    pub const DATA_SIZE: usize = 8;
+}
+
+impl AccountData {
+    /// Version to fill in on new created accounts
+    pub const CURRENT_VERSION: u8 = 1;
+
+    /// Start of writable account data, after version and authority
+    pub const WRITABLE_START_INDEX: usize = 33;
+}
+
+impl IsInitialized for AccountData {
+    /// Is initialized
+    fn is_initialized(&self) -> bool {
+        self.version != 0
```

Comment:
nit: since we set `self.version` explicitly to CURRENT_VERSION during Initialize, aligning the initialized check here seems nice:
```suggestion
        self.version == CURRENT_VERSION
```

Reply:
Done in 0d65c8d